lib/binary/serializer.cpp

```diff
@@ -1,10 +1,17 @@
 #include <kllvm/binary/serializer.h>
 
+#include <arpa/inet.h>
 #include <cassert>
 #include <limits>
 
 namespace kllvm {
 
+namespace detail {
+bool is_big_endian() {
+  return htonl(365) == 365;
```

Comment:
I am not very familiar with the `arpa/inet.h` header but it seems to be OS specific. An easier way to determine endianess is something like this:
```
inline bool is_big_endian() {
  uint32_t i = 1;
  uint8_t *c = (uint8_t*)&i;
  if (*c)
        return false;
    else
        return true;
}
```
Another advantage of this code is that a modern compiler like GCC and LLVM is fairly certain to inline the code and optimize away the branch.

Reply:
That makes sense - thanks!